### ISSUE TITLE
flatten on SCollection[Iterable[Int]] does not work on 0.7.0-beta1

### ISSUE THREAD

**Issue by pishen** [spotify/scio#1551]
The following code compiles without error in Scio 0.6.1
```scala
sc.parallelize(Seq(1,2,3,4,5)).top(3).flatten
```
But when using Scio 0.7.0-beta1, I met the following error:
```
[error] Main.scala:20:43: ambiguous implicit values:
[error]  both method jBitSetCoder in trait JavaCoders of type => com.spotify.scio.coders.Coder[java.util.BitSet]
[error]  and value jShortCoder in trait JavaCoders of type => com.spotify.scio.coders.Coder[Short]
[error]  match expected type com.spotify.scio.coders.Coder[U]
[error]     sc.parallelize(Seq(1,2,3,4,5)).top(3).flatten
[error]                                           ^
[error] one error found
```

**Comment by regadas** [comment]
Hi @pishen try this:

```
sc.parallelize(Seq(1,2,3,4,5)).top(3).flatten[Int]
```

**Comment by mpasa** [comment]
Flatmap works as expected:

```scala
sc.parallelize(Seq(1,2,3,4,5)).top(3).flatMap(identity)
```

So it seems to be a type inference problem with `T => TraversableOnce[U]` and `Coder[U]`. A workaround is to use an implicit parameter for the coder instead of a context bound, but I'm not sure why that works. We can discuss it in this PR #1552.

**Comment by jto** [comment]
Fixed by #1552
